apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/DatastoreTransactionImpl.cs

```diff
@@ -123,6 +123,79 @@ public override AsyncLazyDatastoreQuery RunQueryLazilyAsync(GqlQuery gqlQuery, C
         }
 
+        /// <inheritdoc/>
+        public override AggregationQueryResults RunAggregationQuery(AggregationQuery query, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                AggregationQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions
+
+            };
+            var runAggregationQueryResponse = _client.RunAggregationQuery(request, callSettings);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        /// <inheritdoc/>
+        public override AggregationQueryResults RunAggregationQuery(GqlQuery query, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                GqlQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions,
+
+            };
+            var runAggregationQueryResponse = _client.RunAggregationQuery(request, callSettings);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<AggregationQueryResults> RunAggregationQueryAsync(AggregationQuery query, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                AggregationQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions
+
+            };
+            var runAggregationQueryResponse = await _client.RunAggregationQueryAsync(request, callSettings).ConfigureAwait(false);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<AggregationQueryResults> RunAggregationQueryAsync(GqlQuery query, CallSettings callSettings = null)
+        {
+            var request = new RunAggregationQueryRequest
+            {
+                GqlQuery = query,
+                ProjectId = _projectId,
+                PartitionId = _partitionId,
+                ReadOptions = _readOptions,
+
+            };
+            var runAggregationQueryResponse = await _client.RunAggregationQueryAsync(request, callSettings).ConfigureAwait(false);
+            return ProcessAggregationQueryResponse(runAggregationQueryResponse);
+        }
+
+        internal AggregationQueryResults ProcessAggregationQueryResponse(RunAggregationQueryResponse runAggregationQueryResponse)
+        {
+            var results = new AggregationQueryResults(runAggregationQueryResponse.Batch.MoreResults, runAggregationQueryResponse.Batch.ReadTime);
+            foreach (var aggregationResults in runAggregationQueryResponse.Batch.AggregationResults)
```

Comment:
aggregationResult instead of aggregationResults?

Reply:
This looks like entirely duplicate code, unless I've missed something. If you move this logic into AggregationQueryResults, you can use it from both places.

Reply:
Make sense Jon will update it